BSP/Inc/BSP_I2C.h

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2018-2023 UT Longhorn Racing Solar */
+// BSP I2C header file
+
+#ifndef BSP_I2C_H
+#define BSP_I2C_H
+
+#include <stdint.h>
+#include "BSP.h"
+#include "stm32f4xx_hal.h"
+#include "stm32f4xx_hal_i2c.h"
+#include "stm32f4xx_hal_rcc.h"
+#include "stm32f4xx_hal_rcc_ex.h"
+#include "stm32f4xx_hal_gpio.h"
+
+/**
+ * @brief Initialize I2C port for EEPROM interaction
+ * 
+ */
+BSP_Status BSP_I2C_Init();
+/**
+ * @brief   Transmits data onto the I2C bus.
+ * @param   deviceAddr : the device address to write to
+ * @param   regAddr : the register address to write to in the IC's memory.
+ * @param   txData : the data array to be sent onto the bus.
+ * @param   txLen : the length of the data array.
+ * @return  error status
+ */
+BSP_Status BSP_I2C_Write(uint8_t deviceAddr, uint16_t regAddr, uint8_t *txData, uint32_t txLen);
```

Comment:
Function names in the .c file do not match with functions in the header. Make sure they match up. Add new function declarations in the .h file to match the new functions written